registry/auth/token/util.go

```diff
@@ -36,3 +46,42 @@ func containsAny(ss []string, q []string) bool {
 
 	return false
 }
+
+// NOTE: RFC7638 does not prescribe which hashing function to use, but suggests
+//
+//	sha256 as a sane default as of time of writing
+func hashAndEncode(payload string) string {
+	shasum := sha256.Sum256([]byte(payload))
+	return base64.RawURLEncoding.EncodeToString(shasum[:])
+}
+
+// RFC7638 states in section 3 sub 1 that the keys in the JSON object payload
+// are required to be ordered lexicographical order. Golang does not guarantee
+// order of keys[0]
+// [0]: https://groups.google.com/g/golang-dev/c/zBQwhm3VfvU
+//
+// The payloads are small enough to create the JSON strings manually
+func RFC7638Thumbprint(publickey crypto.PublicKey) string {
```

Comment:
I am not sure about the name of this function 🤔 

As a variable name for a non-func variable, the noun nature of it would make sense (Go standard library has these formatting strings like `time.RFC3339Nano`), but as a name for a func I'm not sure. Maybe prefix it with `Get` or some such?

Reply:
I like the idea of `Get...`, will go that way

Reply:
Renamed as commented!